pkg/types/config.go

```diff
@@ -289,6 +291,16 @@ type EC2ProviderConfig struct {
 	Agent        ProviderAgentConfig `key:"agent" json:"agent"`
 }
 
+type AgentRedisConfig struct {
+	Hostname string `key:"hostname" json:"hostname"`
+}
+
+type RootAgentConfig struct {
+	ControlPlaneRedis AgentRedisConfig `key:"controlPlaneRedis" json:"control_plane_redis"`
+	JuicefsRedis      AgentRedisConfig `key:"juicefsRedis" json:"juicefs_redis"`
```

Comment:
Not sure I like this config structure. We're basically only using it to override the URLs, feels like there's a more elegant approach here

Reply:
i'm not a fan either. especially since we don't pass this to the agent.

we do this
`remoteConfig.Database.Redis.Addrs[0] = fmt.Sprintf("%s:%d", redisHostname, 6379)`
`remoteConfig.Storage.JuiceFS.RedisURI = fmt.Sprintf("%s://:%s@%s", scheme, juicefsRedisPassword, juiceFsRedisHostname)`
`remoteConfig.BlobCache.BlobFs.Sources[idx].JuiceFS.RedisURI = remoteConfig.Storage.JuiceFS.RedisURI`

Reply:
I added a new variable config and removed the juicefs redis, controlplane redis and blobcache urls. 
The new veriable is just called 
`DynamicServiceHosts`

that encapsulate all of them